### ISSUE TITLE
preparePayment returning malformed Payment?

### ISSUE THREAD

**Issue by jefftrudeau** [ripple/ripple-lib#954]
I'm trying to test some basic operations on the test network, wss://s.altnet.rippletest.net:51233. This specific test involves calls to `preparePayment()`, `sign()`, `submit()` and `getTransaction()`. It appears that `preparePayment` is including the `SendMax` attribute regardless of whether `source.maxAmount` or `destination.minAmount` is supplied - is this the intended behavior?

```
const payment = {
  source: {
    address: src.address,
    maxAmount: {
      currency: 'drops',
      value: '1000000',
    },
  },
  destination: {
    address: dst.address,
    amount: {
      currency: 'drops',
      value: '1000000',
    },
  },
};

return api.preparePayment(src.address, payment).then((prepareResult) => {
  const signResult = api.sign(prepareResult.txJSON, src.secret);

  return api.submit(signResult.signedTransaction).then((submitResult) => {
    console.log('Submitted Transaction', submitResult);
  });
}).catch((error) => {
  console.log('Error', error);
});
```

Results in:
`Error [RippledError(Submit failed, { resultCode: 'temBAD_SEND_XRP_MAX',
  resultMessage: 'Malformed: Send max is not allowed for XRP to XRP.' })]`

However, if I do something like the following before calling `sign()`, everything works:

```
if (prepareResult.txJSON.indexOf('SendMax') > -1) {
  const txJSON = JSON.parse(prepareResult.txJSON);
  delete txJSON.SendMax;
  prepareResult.txJSON = JSON.stringify(txJSON);
}
```

**Comment by intelliot** [comment]
@jefftrudeau This is a bug and will be fixed shortly. Thanks for catching this!

**Comment by intelliot** [comment]
Released in 1.0.2
